test/conformance/header_test.go

```diff
@@ -0,0 +1,114 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"knative.dev/eventing/test/base/resources"
+	"knative.dev/eventing/test/common"
+)
+
+const (
+	userHeaderKey   = "this-was-user-set"
+	userHeaderValue = "a value"
+)
+
+// The Channel MUST pass through all tracing information as CloudEvents attributes
+func TestMustPassTracingHeaders(t *testing.T) {
+	singleEvent(t, resources.CloudEventEncodingBinary)
+}
+
+/*
+singleEvent tests the following scenario:
+
+EventSource ---> Channel ---> Subscription ---> Service(Logger)
+
+*/
+func singleEvent(t *testing.T, encoding string) {
```

Comment:
Since this test will also be run in `eventing-contrib`, please put it as a separate helper function and make it public, like what is done for e2e test. This will enable `eventing-contrib` to import it and construct the test.

Please lMK if unclear.

Reply:
No clear or that - do you have example how eventing-contrib is importing to construct test?

Reply:
My PR in `eventing-contrib` is being blocked by the broken dependencies of `eventing` :( Anyway...

- An example of the helper function in `eventing` is https://github.com/knative/eventing/blob/master/test/e2e/helpers/channel_single_event_helper.go.
- An example of the way to call the helper in `eventing-contrib` is https://github.com/knative/eventing-contrib/blob/ece2733bb345e8e40e66cdb61c3c0f55d97fe6fe/test/e2e/channle_single_event_test.go

Reply:
I like this approach, @Fredy-Z

Reply:
If I understand correctly what is needed is to make singleEvent into SingleEventTestHeadersChannelHelper so can be imported by eventing-contrib?

Reply:
Yes. And due to how `dep` works, you'll also have to create a `helpers` package like the one in e2e test - https://github.com/knative/eventing/tree/master/test/e2e/helpers

Reply:
@Fredy-Z made changes following helpers package pattern